test/uk/gov/hmrc/personaldetailsvalidation/formats/PersonalDetailsValidationFormatSpec.scala

```diff
@@ -36,6 +38,81 @@ class PersonalDetailsValidationFormatSpec
+        val personalDetails : PersonalDetailsWithNino = ObjectGenerators.personalDetailsObjects.generateOne.asInstanceOf[PersonalDetailsWithNino]
+        val personalDetailsWithPostCodeJson = new PersonalDetailsWithPostCode(
+          personalDetails.firstName,
+          personalDetails.lastName,
+          personalDetails.dateOfBirth,
+          postCode
+        ).toJson.toString()
+        Json.parse(personalDetailsWithPostCodeJson).as[PersonalDetails]
+      }
+    }
+
+    "correctly fail to parse valid postcodes that do not start at the beginning of the field" in new Setup {
+      import PersonalDetailsFormat._
+      forAll(PostCodeGenerator.postCode) { postCode =>
+        val personalDetails : PersonalDetailsWithNino = ObjectGenerators.personalDetailsObjects.generateOne.asInstanceOf[PersonalDetailsWithNino]
+        val personalDetailsWithPostCodeJson = new PersonalDetailsWithPostCode(
+          personalDetails.firstName,
+          personalDetails.lastName,
+          personalDetails.dateOfBirth,
+          " " + postCode
+        ).toJson.toString()
+        an [JsResultException] should be thrownBy
+          Json.parse(personalDetailsWithPostCodeJson).as[PersonalDetails]
+      }
+    }
+
+    "correctly fail to parse valid postcodes that do not end the field" in new Setup {
+      import PersonalDetailsFormat._
+      forAll(PostCodeGenerator.postCode) { postCode =>
+        val personalDetails : PersonalDetailsWithNino = ObjectGenerators.personalDetailsObjects.generateOne.asInstanceOf[PersonalDetailsWithNino]
+        val personalDetailsWithPostCodeJson = new PersonalDetailsWithPostCode(
+          personalDetails.firstName,
+          personalDetails.lastName,
+          personalDetails.dateOfBirth,
+          postCode + " "
+        ).toJson.toString()
+        an [JsResultException] should be thrownBy
+          Json.parse(personalDetailsWithPostCodeJson).as[PersonalDetails]
+      }
+    }
+
+    "correctly fail to parse the field if it contains multiple postcode" in new Setup {
+      import PersonalDetailsFormat._
+      forAll(PostCodeGenerator.postCode) { postCode =>
+        val secondPostCode = PostCodeGenerator.postCode.generateOne
+        val personalDetails : PersonalDetailsWithNino = ObjectGenerators.personalDetailsObjects.generateOne.asInstanceOf[PersonalDetailsWithNino]
+        val personalDetailsWithPostCodeJson = new PersonalDetailsWithPostCode(
+          personalDetails.firstName,
+          personalDetails.lastName,
+          personalDetails.dateOfBirth,
+          postCode + " " + secondPostCode
+        ).toJson.toString()
+        an [JsResultException] should be thrownBy
+          Json.parse(personalDetailsWithPostCodeJson).as[PersonalDetails]
+      }
+    }
+
+    "fail validation for invalid postcodes" in new Setup {
+      import PersonalDetailsFormat._
+      val invalidPostCode = listOfN[Char](7, frequency((1,choose(65.toChar, 90.toChar)), (1, choose(97.toChar, 122.toChar)))).map(_.mkString)
```

Comment:
Can you see if it is possible to create an alphanumeric string which does not satisfy postcode generator?